src/query-executor/query-executor-base.ts

```diff
@@ -14,6 +14,9 @@ import { Deferred } from '../util/deferred.js'
 
 const NO_PLUGINS: ReadonlyArray<KyselyPlugin> = freeze([])
 
+// TODO: remove.
+let warnedOfDeprecation: boolean
```

Comment:
I think we should add a helper function like logDeprected (or something) in its own module that keeps a Set<string> of deprection messages that have been shown.

Reply:
Or even more generic `logOnce`

Reply:
Roger!